cpp/CMakeLists.txt

```diff
@@ -100,96 +105,141 @@ endif()
 # add third party dependencies using CPM
 rapids_cpm_init()
 
-# thrust and libcudacxx need to be before cuco!
-include(cmake/thirdparty/get_thrust.cmake)
+# cuco before rmm to properly configure thrust/cub
```

Comment:
@cjnolet We should restore the call to `include(cmake/thirdparty/get_thrust.cmake)` as the first line before cuco or rmm.

If we have cuco as the first entry it will bring in thrust 1.12 which we don't want. By having our `thrust` first that will fetch 1.15 and will stop cuco from fetching 1.12

Reply:
@robertmaynard, I need some help here then. 

Without this change, I get a thrust error when building cuml (locally) about the incorrect cub verison being picked up (1.12). With this change, cuml builds (locally) successfully and I believe it's pulling in thrust 1.15 because I would have expected to get a cub build error in HDBSCAN if it were not.

Reply:
@trxcllnt also opened a PR to cuco to update the thrust version to 1.15, which we're using in this PR: https://github.com/NVIDIA/cuCollections/pull/134

Reply:
yes for rmm it uses the version specified by `versions.json` in rapids-cmake